components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java

```diff
@@ -808,6 +808,19 @@ public boolean isAPIAvailable(APIIdentifier identifier) throws APIManagementExce
         }
     }
 
+    public boolean isAPIProductAvailable(APIProductIdentifier identifier) throws APIManagementException {
+        String path = APIConstants.API_ROOT_LOCATION + RegistryConstants.PATH_SEPARATOR +
+                identifier.getProviderName() + RegistryConstants.PATH_SEPARATOR +
+                identifier.getName() + RegistryConstants.PATH_SEPARATOR + identifier.getVersion();
+        try {
+            return registry.resourceExists(path);
+        } catch (RegistryException e) {
+            String msg = "Failed to check availability of API Product :" + path;
+            log.error(msg, e);
```

Comment:
Let's remove log.error. We shouldn't log and throw which is an antipattern.

Reply:
Sure, the same procedure has been done in isAPIAvailable function in line numbers 805-807. Shall I change that too?

Reply:
Yes please, let's do this for other applicable places too,